PhysicsTools/NanoAOD/plugins/BJetEnergyRegressionVarProducer.cc

```diff
@@ -0,0 +1,437 @@
+//     srcMu_(consumes<edm::View<pat::Muon>>(iConfig.getParameter<edm::InputTag>("musrc"))),
+//     srcEle_(consumes<edm::View<pat::Electron>>(iConfig.getParameter<edm::InputTag>("elesrc"))),
+    srcVtx_(consumes<std::vector<reco::Vertex>>(iConfig.getParameter<edm::InputTag>("pvsrc"))),
+    srcSV_(consumes<edm::View<reco::VertexCompositePtrCandidate>>(iConfig.getParameter<edm::InputTag>("svsrc"))),
+	srcGP_(consumes<std::vector<reco::GenParticle>>(iConfig.getParameter<edm::InputTag>("gpsrc")))
+  {
+    //un prodotto da copiare
+    produces<edm::ValueMap<float>>("leptonPtRel");
+    produces<edm::ValueMap<float>>("leptonPtRatio");
+    produces<edm::ValueMap<float>>("leptonPtRelInv");//una variabile sbagliata?
+    produces<edm::ValueMap<float>>("leptonPtRelv0");
+    produces<edm::ValueMap<float>>("leptonPtRatiov0");
+    produces<edm::ValueMap<float>>("leptonPtRelInvv0");//v0 ~ heppy?
+    produces<edm::ValueMap<float>>("leptonPt");
+    produces<edm::ValueMap<int>>("leptonPdgId");
+    produces<edm::ValueMap<float>>("leptonDeltaR");
+    produces<edm::ValueMap<float>>("leadTrackPt"); 
+    produces<edm::ValueMap<float>>("vtxPt");
+    produces<edm::ValueMap<float>>("vtxMass");
+    produces<edm::ValueMap<float>>("vtx3dL");
+    produces<edm::ValueMap<float>>("vtx3deL");
+    produces<edm::ValueMap<int>>("vtxNtrk");
+    produces<edm::ValueMap<float>>("ptD");
+	produces<edm::ValueMap<float>>("genPtwNu");
+    
+    
+  }
+  ~BJetEnergyRegressionVarProducer() override {};
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+   private:
+  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+  
+  std::tuple<float,float,float> calculatePtRatioRel(edm::Ptr<reco::Candidate> lep, edm::Ptr<pat::Jet> jet) const;
+  std::tuple<float,float,float> calculatePtRatioRelSimple(edm::Ptr<reco::Candidate> lep, edm::Ptr<pat::Jet> jet) const; //old version?
+
+        // ----------member data ---------------------------
+
+  edm::EDGetTokenT<edm::View<pat::Jet>> srcJet_;
+//   edm::EDGetTokenT<edm::View<pat::Electron>> srcEle_;
+//   edm::EDGetTokenT<edm::View<pat::Muon>> srcMu_;
+  edm::EDGetTokenT<std::vector<reco::Vertex>> srcVtx_;
+  edm::EDGetTokenT<edm::View<reco::VertexCompositePtrCandidate>> srcSV_;
+  edm::EDGetTokenT<std::vector<reco::GenParticle>> srcGP_;
+};
+
+//
+// constants, enums and typedefs
+//
+
+
+//
+// static data member definitions
+//
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+template <typename T>
+void
+BJetEnergyRegressionVarProducer<T>::produce(edm::StreamID streamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const{
+
+  edm::Handle<edm::View<pat::Jet>> srcJet;
+  iEvent.getByToken(srcJet_, srcJet);
+//   edm::Handle<edm::View<pat::Muon>> srcMu;
+//   iEvent.getByToken(srcMu_, srcMu);
+//   edm::Handle<edm::View<pat::Electron>> srcEle;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant